src/projections.jl

```diff
@@ -253,6 +253,106 @@ function projection_on_set(d::DefaultDistance, v::AbstractVector{T}, ::MOI.DualE
     return v + projection_on_set(d, -v, MOI.ExponentialCone())
 end
 
+"""
+    projection_on_set(::DefaultDistance, v::AbstractVector{T}, ::MOI.PowerCone) where {T}
+
+projection of vector `v` on the power cone
+i.e. `K = {(x,y,z) | x^a * y^(1-a) >= |z|, x>=0, y>=0}`.
+
+References:
+* [Differential properties of Euclidean projection onto power cone]
+(https://link.springer.com/article/10.1007/s00186-015-0514-0), Prop 2.2
+"""
+function projection_on_set(::DefaultDistance, v::AbstractVector{T}, s::MOI.PowerCone) where {T}
+    _check_dimension(v, s)
+
+    if _in_pow_cone(v, s)
+        return v
+    end
+    if _in_pow_cone(-v, MOI.dual_set(s))
+        # if in polar cone Ko = -K*
+        return zeros(T, 3)
+    end
+    if abs(v[3]) <= 1e-10
+        return [max(v[1],0), max(v[2],0), 0]
+    end
+
+    _, proj4 = _solve_system_pow_cone(v, s)
+    return proj4
+end
+
+function _in_pow_cone(v::AbstractVector{T}, cone::MOI.PowerCone; tol=1e-10) where {T}
+    α = cone.exponent
+    return v[1] >= 0 && v[2] >= 0 && tol + v[1]^α * v[2]^(1-α) >= abs(v[3])
+end
+
+function _in_pow_cone(v::AbstractVector{T}, cone::MOI.DualPowerCone; tol=1e-10) where {T}
+    α = cone.exponent
+    return (
+        v[1] >= 0 && v[2] >=0 && tol + (v[1])^α * (v[2])^(1-α) >= α^α * (1-α)^(1-α) * abs(v[3])
+    )
+end
+
+"""
+    _solve_system_pow_cone(v::AbstractVector{T}, s::MOI.PowerCone) where {T}
+
+Solves the system in [1, Proposition 2.2] to determine projection.
+Returns tuple `(r, proj)`:
+* `r` such that `Phi(r) = 0` and `0 < r < abs(v[3])`.
+* `proj` is the projection from case 4 of the power cone
+
+References:
+[1]. [Differential properties of Euclidean projection onto power cone]
+(https://link.springer.com/article/10.1007/s00186-015-0514-0), Prop 2.2
+"""
+function _solve_system_pow_cone(v::AbstractVector{T}, s::MOI.PowerCone; max_iters=200, tol=1e-8) where {T}
+    x, y, z = v
+    α = s.exponent
+    Phi_prod(xi,αi,z,r) = max(xi + sqrt(xi^2 + 4*αi*r*(abs(z) - r)), 1e-12)
+    Phi(r) = 0.5*(Phi_prod(x,α,z,r)^α * Phi_prod(y,1-α,z,r)^(1-α)) - r
+    Phi(r,px,py) = 0.5*(px^α * py^(1-α)) - r
+    dPhi_prod_dr(xi,αi,z,r,px) = 2*αi*(abs(z) - 2r) / (px - xi)
+    dPhi_dr(r,phi,px,py,dpx,dpy) = (phi+r)*(α*dpx/px + (1-α)*dpy/py) - 1
+
+    # Solve with Newton method
+    # Start Newton at |z|/2. Sol in set (0, |z|)
+    px, py = zero(T), zero(T)
+    r = abs(z) / 2
+    for ii in 1:max_iters
+        px = Phi_prod(x, α, z, r)
+        py = Phi_prod(y, 1-α, z, r)
+        phi = Phi(r, px, py)
+
+        abs(phi) < tol && break
+
+        dpx = dPhi_prod_dr(x,α,z,r,px)
+        dpy = dPhi_prod_dr(y,1-α,z,r,py)
+        dphi = dPhi_dr(r,phi,px,py,dpx,dpy)
+
+        # Newton step, bounded to interval
+        r = min(max(r - phi/dphi, 0), abs(z))
+
+        ii == max_iters && @warn("Maximum iterations hit on power cone proj")
+    end
+
+    return r, [0.5*px; 0.5*py; sign(z)*r]
```

Comment:
to apply only if all three terms are scalar, which I think is the case here
```suggestion
    return r, [0.5*px, 0.5*py, sign(z)*r]
```

Reply:
Changed